telekom-workflow-engine/src/main/java/ee/telekom/workflow/core/archive/ArchiveDao.java

```diff
@@ -15,21 +16,36 @@
 @Repository
 public class ArchiveDao extends AbstractWorkflowEngineDao{
 
-    public void archive( long woinRefNum ){
-        createArchiveWorkflowInstance( woinRefNum );
+    public void archive( long woinRefNum, int archivePeriodLength ){
+        createArchiveWorkflowInstance( woinRefNum, archivePeriodLength );
         createArchiveWorkItems( woinRefNum );
         deleteOriginalWorkItems( woinRefNum );
         deleteOriginalWorkflowInstance( woinRefNum );
     }
 
-    private void createArchiveWorkflowInstance( long woinRefNum ){
+    public void cleanup(){
+        List<Long> refNums = getJdbcTemplate().queryForList( "SELECT ref_num FROM " + getSchema() + "workflow_instances_archive WHERE cleanup_after IS NOT NULL AND cleanup_after <= CURRENT_TIMESTAMP", Long.class );
+        if (!refNums.isEmpty()) {
+            AdvancedParameterSource[] sources = new AdvancedParameterSource[refNums.size()];
+            for( int i = 0; i < refNums.size(); i++ ){
+                sources[i] = new AdvancedParameterSource().addValue( "refNum", refNums.get(i) );
+            }
+            getNamedParameterJdbcTemplate().batchUpdate("DELETE FROM " + getSchema() + "work_items_archive WHERE woin_ref_num IN (:refNum)", sources);
+            getNamedParameterJdbcTemplate().batchUpdate("DELETE FROM " + getSchema() + "workflow_instances_archive WHERE ref_num IN (:refNum)", sources);
```

Comment:
a little sus

The initial select could return 200k refNums. Then it builds 200k sources, one per each refNum? And then batch-runs 200k queries with one refNum per query? Then the syntax ref_num IN (:refNum) doesn't make sense, if there is always 1 refNum in the actual value?

Or maybe I'm not understanding how it runs.